File: C/Q42.c
#include <stdio.h>

//Function to calculate power using recursion
int power(int base, int exp) {
    if (exp == 0) // anything raised to 0 is 1
        return 1;             
    else
        return base * power(base, exp - 1); 
}

int main() {
    int base, exp;
    printf("Enter base: ");
    scanf("%d", &base);
    printf("Enter exponent: ");
    scanf("%d", &exp);

    printf("%d^%d = %d\n", base, exp, power(base, exp));

    return 0;
}

File: Python/Q42.py
# Q42.py - Program to Calculate Power Using Recursion
def power(base, exp):
    if exp == 0:
        return 1
    return base * power(base, exp-1)

b = float(input("Enter base: "))
e = int(input("Enter exponent: "))
print(f"Result: {power(b, e)}")